### PR TITLE
Add filter to narrow down mutations and improve performance

### PR DESCRIPTION
**Pull request by wongyouth**
Right now every mutations will kick off the trigger to save state to storage even though the mutation will not change the persisted state. It does not perform very well in a large project.
Adding a `filter` parameter to filter mutations only related to persisted state can improve performance.

### CODE REVIEW

**Comment by robinvdvleuten** [comment]
@wongyouth good idea! Thanks for the PR 👍

File: README.md [hunk 1]
```diff
@@ -37,6 +37,7 @@ can be provided to configure the plugin for your specific needs:
 - `storage <Storage>`: Instead for (or in combination with) `getState` and `setState`. Defaults to localStorage (or internal storage for Server Side Rendering).
 - `getState <Function>`: A function that will be called to rehydrate a previously persisted state. Defaults to using `storage`.
 - `setState <Function>`: A function that will be called to persist the given state. Defaults to using `storage`.
+- `filter <Function>`: A function that will be called to filter to specific mumations with which will trigger `setState` on storage. Defaults to `() => true`
```
Comment (robinvdvleuten) [inline README.md]: mumations => mutations.

Comment (robinvdvleuten) [inline README.md]: Better would be "A function that will be called to filter any mutations which will trigger `setState` on storage eventually."

Comment (wongyouth) [inline README.md]: Thanks for your review. I will fix this.

**Comment by robinvdvleuten** [comment]
Thanks @wongyouth. I've merged it and will make a new release soon 👍

**Comment by wongyouth** [comment]
👍

**Pull request merged**